packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+    uint256 updatedDelegatedAmount = ((value * 2) / 100) * percent;
+
+    vm.prank(delegator);
+    lockedGold.delegateGovernanceVotes(
+      delegatee,
+      FixidityLib.newFixedFraction(percent, 100).unwrap()
+    );
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+
+    (uint256 expectedAmount, uint256 realAmount) = lockedGold
+      .getDelegatorDelegateeExpectedAndRealAmount(delegator, delegatee);
+    assertEq(expectedAmount, updatedDelegatedAmount);
+    assertEq(realAmount, updatedDelegatedAmount);
+
+    assertEq(
+      lockedGold.getAccountTotalGovernanceVotingPower(delegatee),
+      updatedDelegatedAmount + value
+    );
+  }
+
+  function test_ShouldReturnEqualAmounts_WhenDelegated_WhenVOteSigners() public {
+    helper_WhenVoteSigners();
+    uint256 percent = 70;
+    uint256 amount = (value / 100) * percent;
+
+    vm.prank(delegatorSigner);
+    lockedGold.delegateGovernanceVotes(
+      delegatee,
+      FixidityLib.newFixedFraction(percent, 100).unwrap()
+    );
+
+    (uint256 expectedAmount, uint256 realAmount) = lockedGold
+      .getDelegatorDelegateeExpectedAndRealAmount(delegator, delegatee);
+    assertEq(expectedAmount, amount);
+    assertEq(realAmount, amount);
+  }
+
+  function test_ShouldReturnEqualAmountAndUpdateTotalVotingPowerOfDelegatee_WhenMoreCeloLocked_WhenVoteSigners()
+    public
+  {
+    helper_WhenVoteSigners();
+    uint256 percent = 70;
+    uint256 updatedDelegatedAmount = ((value * 2) / 100) * percent;
+
+    vm.prank(delegatorSigner);
+    lockedGold.delegateGovernanceVotes(
+      delegatee,
+      FixidityLib.newFixedFraction(percent, 100).unwrap()
+    );
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+
+    (uint256 expectedAmount, uint256 realAmount) = lockedGold
+      .getDelegatorDelegateeExpectedAndRealAmount(delegator, delegatee);
+    assertEq(expectedAmount, updatedDelegatedAmount);
+    assertEq(realAmount, updatedDelegatedAmount);
+
+    assertEq(
+      lockedGold.getAccountTotalGovernanceVotingPower(delegatee),
+      updatedDelegatedAmount + value
+    );
+  }
+}
+
+contract UpdateDelegatedAmount is LockedGoldFoundryTest {
+  address delegator = actor("delegator");
+  address delegatee = actor("delegatee");
+  address delegatorSigner;
+  uint256 delegatorSignerPK;
+  address delegateeSigner;
+  uint256 delegateeSignerPK;
+  uint256 value = 1000;
+  uint256 delegatedPercent = 70;
+  uint256 delegatedAmount = (value / 100) * delegatedPercent;
+
+  function setUp() public {
+    super.setUp();
+
+    vm.prank(delegator);
+    accounts.createAccount();
+    vm.prank(delegatee);
+    accounts.createAccount();
+
+    vm.deal(delegator, 10 ether);
+    vm.deal(delegatee, 10 ether);
+
+    vm.prank(delegator);
+    lockedGold.lock.value(value)();
+
+    (delegatorSigner, delegatorSignerPK) = actorWithPK("delegatorSigner");
+    (delegateeSigner, delegateeSignerPK) = actorWithPK("delegateeSigner");
+  }
+
+  function helper_WhenVoteSigners() public {
```

Comment:
This function is also repeated and could go to the parent or an abstract contract.